clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoClientBase.java

```diff
@@ -37,13 +42,59 @@ public abstract class GravitinoClientBase implements Closeable {
    *
    * @param uri The base URI for the Gravitino API.
    * @param authDataProvider The provider of the data which is used for authentication.
+   * @param checkVersion Whether to check the version of the Gravitino server.
    */
-  protected GravitinoClientBase(String uri, AuthDataProvider authDataProvider) {
-    this.restClient =
-        HTTPClient.builder(Collections.emptyMap())
-            .uri(uri)
-            .withAuthDataProvider(authDataProvider)
-            .build();
+  protected GravitinoClientBase(
+      String uri, AuthDataProvider authDataProvider, boolean checkVersion) {
+    ObjectMapper mapper = JsonUtils.objectMapper();
+    mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+
+    if (checkVersion) {
+      this.restClient =
+          HTTPClient.builder(Collections.emptyMap())
+              .uri(uri)
+              .withAuthDataProvider(authDataProvider)
+              .withObjectMapper(mapper)
+              .withPreConnectHandle(
+                  (client) -> {
+                    checkVersion();
+                    return null;
+                  })
+              .build();
+
+    } else {
+      this.restClient =
+          HTTPClient.builder(Collections.emptyMap())
+              .uri(uri)
+              .withAuthDataProvider(authDataProvider)
+              .withObjectMapper(mapper)
+              .build();
+    }
+  }
+
+  /**
+   * Check the compatibility of the client with the target server.
+   *
+   * @throws GravitinoRuntimeException If the client version is greater than the server version.
+   */
+  public void checkVersion() {
+    GravitinoVersion serverVersion = getVersion();
```

Comment:
You'd better make the current `getVersion` as deprecated, and create new API called `serverVersion`. Also point the implementation of `getVersion` to `serverVersion`.